tree/ntuple/v7/test/ntuple_processor_join.cxx

```diff
@@ -81,23 +81,14 @@ class RNTupleJoinProcessorTest : public testing::Test {
    }
 };
 
-TEST_F(RNTupleJoinProcessorTest, Basic)
+TEST_F(RNTupleJoinProcessorTest, PrimaryOnly)
 {
-   std::vector<RNTupleOpenSpec> ntuples;
-   try {
-      auto proc = RNTupleProcessor::CreateJoin(ntuples, {});
-      FAIL() << "creating a processor without at least one RNTuple should throw";
-   } catch (const ROOT::RException &err) {
-      EXPECT_THAT(err.what(), testing::HasSubstr("at least one RNTuple must be provided"));
-   }
-
-   ntuples = {{fNTupleNames[0], fFileNames[0]}};
-
-   auto proc = RNTupleProcessor::CreateJoin(ntuples, {});
+   // Primary ntuple only
+   auto proc = RNTupleProcessor::CreateJoin({fNTupleNames[0], fFileNames[0]}, {}, {});
```

Comment:
I know this test was already here before, but looking at it again made me wonder. Is this just a sanity check or do we expect this syntax to be used? On the surface it looks weird that a user calls `CreateJoin` if there is only one RNTuple.

Reply:
I don't know if we *expect* this syntax to be used, but I don't see why we shouldn't allow this use case (we're also not disallowing creating a chain with only one ntuple). I could think of a use case where users more or less automatize the creation of the processor and then having this possibility could be useful. I don't think it matters much for performance because practically all `auxNTuple` stuff is just skipped, but this is of course something we can benchmark to make sure.